### DISCUSSION TITLE
Couldn't do a hot update of nats.config

### DISCUSSION THREAD

**Discussion by mgonzalez-etra** [nats-io/nats-server#2964] [General]
We have a docker container with the nats image (we tryed with alpine and linux versions) and the command for changing the config without restarting the container didn't work. As there is no bash (or didn't find) I used the following command: 

`docker exec -ti nats_server.imagekey /nats-server --signal reload`

The response from the container was: 

"nats-server: unable to resolve pid, try providing one"

Did I miss something in the command or is it some kind of bug?

As the intention was to just change the nats users, there was no need to restart the whole container but we found no other solution for that. Also, the version of the image was nats and later tryed with nats:linux, all today.

Thats for the attention

**Comment by wallyqs** [comment]
Since there is no bash for `nats-server` to do a `ps` to try to find a running server process in the same namespace then have to set the explicit pid like this `nats-server --signal reload=1` (assuming `1` is the pid in this case).  There is a `-P` flag to set the pidfile as well where you can point to that file instead as well.
